### DISCUSSION TITLE
FR: Give matrix blocks a post- and expiration date/time

### DISCUSSION THREAD

**Discussion by nilsenpaul** [craftcms/cms#8776] [Ideas]
We would like to give our users the option to display a matrix block's content based on a start and end date. This way, it would be possible to show time-sensitive page content without having to manually enable/disable the matrix block it's in.

**Example 1**
I have a page on which I want to display the opening hours during the holiday season. I only want my visitors to see this message from November 30th until January 1th. I add the matrix block, click the little cog on the right and set a post- and expiration date for the block. On the frontend, the block only gets displayed when I want it to.

**Example 2**
When publishing the yearly results, our client wants to show those after a certain date/time.

**Comment by thupsi** [comment]
@nilsenpaul I think this functionality can already be achieved by adding date fields to the matrix block and handling the logic in your templates.
